pkg/cloudprovider/types.go

```diff
@@ -118,6 +122,14 @@ func (ofs Offerings) Get(ct, zone string) (Offering, bool) {
 	})
 }
 
+// Filter filters the offerings based on the passed requirements
+func (ofs Offerings) Filter(requirements scheduling.Requirements) Offerings {
```

Comment:
What about Available?

Reply:
I changed the name to `Requirements`. I saw the `Available` as a different filter so we effectively do

```
instanceType.Offerings.Available().Requirements(req)
```

Reply:
It basically makes it so that we can chain filters in a nice way